### ISSUE TITLE
jupyter python2 notebook may truncate in firefox

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/smc#1537]
Reported on sage-cloud mailing list here, including zip file: https://groups.google.com/d/msg/sage-cloud/b8tgmK8gZ14/5eFs9HFaEQAJ

To reproduce:
- use firefox browser
- upload `Skill4.zip` into an SMC project and unzip
- in a .term, do `wc -c Skill-04-ReadWriteFiles.ipynb`; file length should be 20693
- open `Skill-04-ReadWriteFiles.ipynb`
    - you may see a warning like the screen capture below
    - notebook appears truncated; last cell displayed ends with paragraph
    __Using np.savetxt to Write Data Files ... __
- wait about a minute for the first jupyter auto-save
- in .term, again do `wc -c Skill-04-ReadWriteFiles.ipynb` and note file length is now 4202

Error seen on opening `Skill-04-ReadWriteFiles.ipynb` in firefox:  

<img width="623" alt="screenshot 2017-01-17 12 01 49" src="https://cloud.githubusercontent.com/assets/528072/22033771/6400d330-dcaf-11e6-8435-7738d465ec2e.png">

Workaround 1 - use Google Chrome at least once
- close jupyter notebook and restore `Skill-04-ReadWriteFiles.ipynb` by unzipping again
- open `Skill-04-ReadWriteFiles.ipynb` in chrome
- execute the first cell
- save the notebook
- close the SMC chrome browser tab 
- in firefox session
    - note `wc -c Skill-04-ReadWriteFiles.ipynb` now gives length 13384
    - open jupyter notebook and observe it is no longer truncated

Workaround 2 - convert to .sagews:
- close jupyter notebook and restore `Skill-04-ReadWriteFiles.ipynb` by unzipping again
- convert to sagews with `smc-ipynb2sagews Skill-04-ReadWriteFiles.ipynb`
- open `Skill-04-ReadWriteFiles.sagews` and work with sagews

**Comment by williamstein** [comment]
Interesting: the `fontSize is undefined` error is the same thing that I fixed in https://github.com/sagemathinc/smc/issues/1343

**Comment by williamstein** [comment]
My latest commit seems to fix this -- it's just a worse hack...

**Comment by DrXyzzy** [comment]
Problem is still there for me. Firefox 50.1.0. OS X 10.11.6. I reset my dev project, refreshed Firefox  SMC browser tab, and unzipped a fresh copy of `Skill-04-ReadWriteFiles.ipynb`. 

Screen capture shows where the notebook is truncated as indicated by drop shadow below `In [3]:` display.
<img width="637" alt="screenshot 2017-01-18 04 48 12" src="https://cloud.githubusercontent.com/assets/528072/22061196/7d5aafe2-dd39-11e6-89c9-1bf40b9982fe.png">

**Comment by DrXyzzy** [comment]
I see, the commit was reverted.

**Comment by williamstein** [comment]
NOTE: I've completely disabled Jupyter + firefox for now.

**Comment by DrXyzzy** [comment]
As with #1650, I'm marking this blocked until the next release of Firefox (after 51.0.1).

**Comment by DrXyzzy** [comment]
I just tested in Firefox 52.0, getting around the FF block on .ipynb by overriding the user agent setting. Bug is still there, i.e. opening the test notebook mentioned above in FF results in truncation - in fact it is even worse, in that none of the test notebook is visible on opening the file. Character count of `Skill-04-ReadWriteFiles.ipynb` goes from 14913 after unzip to 617 after attempting to open in FF. File opens without truncation in Safari.

**Comment by DrXyzzy** [comment]
Reassigning to @williamstein because it looks as if this issue will unblock when some work in progress with Jupyter is completed.

**Comment by williamstein** [comment]
Closing this in favor of https://github.com/sagemathinc/smc/issues/1754, where Firefox will work perfectly.
